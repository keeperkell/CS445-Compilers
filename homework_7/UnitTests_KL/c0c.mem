Func: input returns type int [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 1  Func: output returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type int [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 2  Func: inputb returns type bool [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 3  Func: outputb returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type bool [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 4  Func: inputc returns type char [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 5  Func: outputc returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type char [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 6  Func: outnl returns type void [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 7  Func: main returns type void [mem: Global loc: 0 size: -2] [line: 3]
.   Child: 1  Compound [mem: None loc: 0 size: -5] [line: 4]
.   .   Child: 0  Var: x of type int [mem: Local loc: -2 size: 1] [line: 5]
.   .   Sibling: 1  Var: y of type int [mem: Local loc: -3 size: 1] [line: 6]
.   .   Sibling: 2  Var: z of type int [mem: Local loc: -4 size: 1] [line: 7]
.   .   Child: 1  Assign: <- of type int [line: 9]
.   .   .   Child: 0  Id: x of type int [mem: Local loc: -2 size: 1] [line: 9]
.   .   .   Child: 1  Const 753 of type int [line: 9]
.   .   Sibling: 1  Assign: <- of type int [line: 10]
.   .   .   Child: 0  Id: y of type int [mem: Local loc: -3 size: 1] [line: 10]
.   .   .   Child: 1  Const 444 of type int [line: 10]
.   .   Sibling: 2  Call: output of type void [line: 11]
.   .   .   Child: 0  Assign: <- of type int [line: 11]
.   .   .   .   Child: 0  Id: z of type int [mem: Local loc: -4 size: 1] [line: 11]
.   .   .   .   Child: 1  Assign: <- of type int [line: 11]
.   .   .   .   .   Child: 0  Id: x of type int [mem: Local loc: -2 size: 1] [line: 11]
.   .   .   .   .   Child: 1  Id: y of type int [mem: Local loc: -3 size: 1] [line: 11]
.   .   Sibling: 3  Call: outnl of type void [line: 12]
Offset for end of global space: 0
Number of warnings: 0
Number of errors: 0
