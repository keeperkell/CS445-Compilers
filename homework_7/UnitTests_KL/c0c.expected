Number of warnings: 0
Number of errors: 0
------------------------------------
Loading file: UnitTests/c0c.tm
444
Bye.
